cmd/genji/shell/shell.go

```diff
@@ -370,6 +385,20 @@ func (sh *Shell) getAllTables() ([]string, error) {
 }
 
 func (sh *Shell) completer(in prompt.Document) []prompt.Suggest {
+	if strings.HasPrefix(in.Text, ".") {
+		// This check avoids infinite suggestions.
+		if isCommand(in.Text) {
```

Comment:
This is necessary because you are using `GetWordBeforeCursor` below. If you change it to `in.Text` instead you won't have this problem.